### PR TITLE
Add support for kanji and kana statement descriptors

### LINKED ISSUE #6874: Support kanji and kana statement descriptors for Japanese merchants
### Description
Additional kanji and kana statement descriptors are supported for Japanese merchants. It is recommended to support Latin, kanji, and kana descriptors to reduce chargeback for Japanese merchants.
For JP Stripe accounts, add statement descriptors for kana and kanji. 
Under the latin statement descriptor, if it's a JP stripe account, we need to show another 2 inputs for kana and kanji statement descriptors.
Max length for kanji - 17 chars
Max length for kana - 22 chars

### PR DESCRIPTION
**Pull request by dpaun1985**
Fixes #6874 

#### Changes proposed in this Pull Request
For JP Stripe accounts, add statement descriptors for kana and kanji. 
Under the latin statement descriptor, if it's a JP stripe account, we need to show another 2 inputs for kana and kanji statement descriptors.
Max length for kanji - 17 chars
Max length for kana - 22 chars

#### Testing instructions

* checkout server PR 3801-gh-Automattic/woocommerce-payments-server 
* use a JP Stripe account
* go to Woo->Settings->Payments
* check `Transactions` section, the inputs for kana and kanji statement descriptors should be present under the latin statement descriptor input
* check for non JP Stripe account. the inputs should not be shown.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7051` or branch name `add/6874-add-kanji-kana` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/6874-add-kanji-kana) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 1596f3ea81b084fff33c8898b00a31461c1a7452
- Build time: 2023-08-31 14:54:21 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/settings/transactions/index.js [hunk 1]
```diff
@@ -94,7 +111,64 @@ const Transactions = ( { setTransactionInputsValid } ) => {
...
+										'woocommerce-payments'
+									) }
+									value={ accountStatementDescriptorKanji }
+									onChange={
+										setAccountStatementDescriptorKanji
+									}
+									maxLength={
+										ACCOUNT_STATEMENT_MAX_LENGTH_KANJI
+									}
+									data-testid={ 'store-name-bank-statement' }
```
Comment (anu-rock) [inline client/settings/transactions/index.js]: We should use unique testid for both the new fields. Eg. `store-name-bank-statement-kanji`.

**Review by anu-rock — COMMENTED** [review]
The code looks good. The happy path worked well in manual testing. One edge case failed.

___

#### Review checklist

| Parameter  | Result |
| ------------- | ------------- |
| Approach  | ✅ All good |
| Implementation  | ✅ All good |
| Automated tests  | ✅ All good |
| Naming  | ✅ All good |
| Typing  | ✅ All good |
| Code comments  | ✅ All good |
| Changelog  | ✅ All good |
| Testing instructions  | 👍 Almost good (we can also add instructions to try saving the fields) |
| Manual testing  | ✅ Works as expected |
| Edge cases  | ❌ Failed (see below) |

#### Scenarios tested

| Scenario  | Result |
| ------------- | ------------- |
| Visibility of Kanji and Kana statement descriptor fields for a JP account  | ✅  Visible |
| Visibility of Kanji and Kana statement descriptor fields for a US account  | ✅  Not visible |
| Save settings on inputting valid values in the new fields  | ✅  Settings saved successfully (verified after page refresh) |
| Save settings on inputting invalid values in the new fields (Latin characters in Kana field)  | ❌  Settings saved successfully (found old values on page refresh). Ideally, an error should be thrown and saving should fail. |

**Comment by dpaun1985** [comment]
Since the texts "Edit the way your store name appears on your customers' bank statements." appears under each input, would be a good ideea to move it on top ? 
Also, for Kanji and Kana inputs, we need only kanji/kana characters. We could add a notice under each input to mention this.
Screenshot 2023-08-29 at 15 38 58

 @elizaan36, could you please share your opinion on this ?

**Review by elazzabi — APPROVED** [review]
Looks good to me and works as expected.

I also agree with Anurag's comment ^ above about having unique ids.

**Review by mordeth — APPROVED** [review]
✅ Code looks good!
✅ Manually tested, and everything works well!

**Comment by elizaan36** [comment]
Moving that line to the top looks good but I think the section title should be "Customer statements". We can keep the "Customer support" section below for email and phone.

image

One small thing "You must only use Kanji letters" is repeated and 2nd one should say Kana. I updated the text to "Use only [language] letters."

Can we detect the store language setting rather than using the word "Latin"? I'm afraid "Latin" won't make much sense to merchants and could be confusing.

File: client/settings/transactions/index.js (hunk 1 above)
Comment (dpaun1985) [reply client/settings/transactions/index.js]: fixed in https://github.com/Automattic/woocommerce-payments/pull/7051/commits/ea43bfa827d90a93b8fd4e388bd0189a1f433de4 thank you!

**Review by anu-rock — APPROVED** [review]
LGTM after latest changes 🎉 

IMO, we should make one last small change. The names for all three fields now read the same, making them appear redundant. I think they should be:

Customer bank statement
Customer bank statement (kanji)
Customer bank statement (kana)

image


Pre-approving the changes ✅ 

(Noting that my edge case issue about the error in saving with incorrect characters will be tackled separately in #7090.)

**Comment by dpaun1985** [comment]
> Moving that line to the top looks good but I think the section title should be "Customer statements". We can keep the "Customer support" section below for email and phone.
> 
> image
> One small thing "You must only use Kanji letters" is repeated and 2nd one should say Kana. I updated the text to "Use only [language] letters."
> 
> Can we detect the store language setting rather than using the word "Latin"? I'm afraid "Latin" won't make much sense to merchants and could be confusing.

@elizaan36 , as you mentioned `%language` should be shown only for Japan. In this case, the message will be "You must only use Japan letters", but based on Stripe [docs](https://stripe.com/docs/payments/account/statement-descriptors#requirements), the field accept only latin letters. 
So, since this is shown only for Japan merchants, we decided to show the text "Use only latin characters" for the default statement descriptor. 
For the other merchants, the message will not be displayed since it's only 1 statement descriptor field.

**Pull request merged**